### PR TITLE
[16.0][MIG] stock_inventory_discrepancy: Migration to version 16.0

### PR DESCRIPTION
**Pull request by carolinafernandez-tecnativa**
Standard migration to v16

@Tecnativa

TT44362

@pedrobaeza

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Remember to squash administrative commits.

**Comment by carolinafernandez-tecnativa** [comment]
> Remember to squash administrative commits.

@pedrobaeza done!

File: stock_inventory_discrepancy/tests/test_inventory_discrepancy.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        #                     "product_id": self.product2.id,
+        #                     "product_uom_id": self.env.ref("uom.product_uom_unit").id,
+        #                     "product_qty": 4.0,
+        #                     "location_id": self.test_loc.id,
+        #                 },
+        #             ),
+        #         ],
+        #     }
+        # )
+        # starting_inv.action_force_done()
```
Comment (sergio-teruel) [inline stock_inventory_discrepancy/tests/test_inventory_discrepancy.py]: Remove commented lines

**Review by sergio-teruel — APPROVED** [review]
Thanks... Only minor comment

File: stock_inventory_discrepancy/tests/test_inventory_discrepancy.py (hunk 1 above)
Comment (pedrobaeza) [reply stock_inventory_discrepancy/tests/test_inventory_discrepancy.py]: @carolinafernandez-tecnativa please attend this

Comment (carolinafernandez-tecnativa) [reply stock_inventory_discrepancy/tests/test_inventory_discrepancy.py]: @pedrobaeza @sergio-teruel done!

**Comment by pedrobaeza** [comment]
The `pre-commit stuff` now contains the changes prior to this last one. I think you have squashed incorrectly.

File: stock_inventory_discrepancy/tests/test_inventory_discrepancy.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    def test_discrepancy_validation(self):
+        """Tests the new workflow"""
+        # quant_line1 is over discrepancy but quant_line2 is not
+        self.quant_line1.write(
+            {
+                "inventory_quantity": 3.0,
+                "inventory_quantity_set": True,
+            }
+        )
+        self.quant_line1._compute_discrepancy_threshold()
```
Comment (AungKoKoLin1997) [inline stock_inventory_discrepancy/tests/test_inventory_discrepancy.py:102]: Calling this compute function after a quant is created will force a call to the write() method, which may cause tests to fail due to the changed behavior of the write() method in this commit: https://github.com/odoo/odoo/commit/e21c17fae8a29c8ee04883b27d761b8ce13cf5b7.

Comment (rousseldenis) [reply stock_inventory_discrepancy/tests/test_inventory_discrepancy.py:102]: Indeed, in test you should set inventory_mode=False in context

**Comment by carolinafernandez-tecnativa** [comment]
ping @pedrobaeza

**Pull request merged**